composeApp/src/desktopMain/kotlin/org/ooni/probe/BuildDependencies.kt

```diff
@@ -48,11 +74,12 @@ val dependencies = Dependencies(
     launchAction = ::launchAction,
     batteryOptimization = object : BatteryOptimization {},
     isWebViewAvailable = { true },
+    isCleanUpRequired = ::hasLegacyDirectories,
+    cleanupLegacyDirectories = ::cleanupLegacyDirectories,
```

Comment:
This methods could be on a separate class/file, to continuously growing the BuildDependencies file. Also, they could be bundled together like on Battery Optimization.